app/lib/service_pattern.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module ServicePattern
```

Comment:
I think we should add this in a separate PR, we already have some service-like classes in `app/lib` so I think if we add this module (which I like) we should also apply it to those classes.